File: test-webhook-server.js
// Simple webhook test server
// Run: node test-webhook-server.js
// Then use ngrok to expose it: npx ngrok http 3002

const express = require('express');
const crypto = require('crypto');

const app = express();
const PORT = 3002;

app.use(express.json());

// Store received webhooks
const receivedWebhooks = [];

// Webhook endpoint
app.post('/webhook', (req, res) => {
  const signature = req.headers['x-webhook-signature'];
  const timestamp = req.headers['x-webhook-timestamp'];
  
  console.log('\n📥 Webhook Received!');
  console.log('Timestamp:', timestamp);
  console.log('Signature:', signature);
  console.log('Body:', JSON.stringify(req.body, null, 2));
  
  // Store webhook
  receivedWebhooks.push({
    timestamp: new Date().toISOString(),
    signature,
    body: req.body
  });
  
  // Always return success for testing
  res.json({ 
    success: true, 
    message: 'Webhook received successfully',
    receivedAt: new Date().toISOString()
  });
});

// View all received webhooks
app.get('/webhooks', (req, res) => {
  res.json({
    count: receivedWebhooks.length,
    webhooks: receivedWebhooks
  });
});

// Clear webhooks
app.delete('/webhooks', (req, res) => {
  receivedWebhooks.length = 0;
  res.json({ message: 'Webhooks cleared' });
});

app.listen(PORT, () => {
  console.log(`\n🚀 Webhook Test Server running on http://localhost:${PORT}`);
  console.log(`📡 Webhook endpoint: http://localhost:${PORT}/webhook`);
  console.log(`\n📋 To expose publicly, run: npx ngrok http ${PORT}`);
  console.log(`   Then use the ngrok URL in your webhook registration\n`);
});

File: test-webhook-server.py
#!/usr/bin/env python3
"""
Simple webhook test server
Run: python3 test-webhook-server.py
Then use ngrok to expose it: ngrok http 3002
"""

from flask import Flask, request, jsonify
from datetime import datetime
import json

app = Flask(__name__)

# Store received webhooks
received_webhooks = []

@app.route('/webhook', methods=['POST'])
def webhook():
    signature = request.headers.get('X-Webhook-Signature')
    timestamp = request.headers.get('X-Webhook-Timestamp')
    
    print('\n📥 Webhook Received!')
    print(f'Timestamp: {timestamp}')
    print(f'Signature: {signature}')
    print(f'Body: {json.dumps(request.json, indent=2)}')
    
    # Store webhook
    received_webhooks.append({
        'timestamp': datetime.now().isoformat(),
        'signature': signature,
        'body': request.json
    })
    
    return jsonify({
        'success': True,
        'message': 'Webhook received successfully',
        'receivedAt': datetime.now().isoformat()
    })

@app.route('/webhooks', methods=['GET'])
def list_webhooks():
    return jsonify({
        'count': len(received_webhooks),
        'webhooks': received_webhooks
    })

@app.route('/webhooks', methods=['DELETE'])
def clear_webhooks():
    received_webhooks.clear()
    return jsonify({'message': 'Webhooks cleared'})

if __name__ == '__main__':
    print('\n🚀 Webhook Test Server running on http://localhost:3002')
    print('📡 Webhook endpoint: http://localhost:3002/webhook')
    print('\n📋 To expose publicly, run: ngrok http 3002')
    print('   Then use the ngrok URL in your webhook registration\n')
    app.run(port=3002, debug=True)

